### ISSUE TITLE
Updating Approval Workflow causes LHN to fall out of sync between devices

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#51349]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @AndrewGable 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1729272573928429

## Action Performed:
Break down in numbered steps

Use the app normally on native ios and on web at the same time throughout the day

## Expected Result:
Describe what you think should've happened

The LHN on both platforms will remain in sync

## Actual Result:
Describe what actually happened

Some reports are showing as unread on mobile while they do not show up in web

Seems like the only difference between the native mobile and web onyx version is that on native the report data includes the `state` property whereas the web version does not.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @cristipaval</details>

**Comment by VickyStash** [comment]
Hi, I'm Viktoryia from Callstack - expert contributor group - and I would like to work on this issue.

**Comment by puneetlath** [comment]
Ah good to know! Are you able to pinpoint any API responses or pusher events that are leading to those mis-matches?

**Comment by VickyStash** [comment]
### Reproduction steps:
**Pre-steps:**
1. Open the **desktop** app and login **User A-Admin**.
2. Open the **web** app and login as **User B**. 
3. Open the **ios** app and login as **User B**, close the ios app.
4. Open the web app in incognito mode and log in as **User C**.

**Steps:**
1. From User A 
1.1 create a workspace
1.2 add User B and User C as members
1.2 enable Workflow feature
1.3 turn on approvals and make the User B approver.
2. From User B submit the expense to the workspace.
3. See that for User B the expense appeared in the LHN.
4. From User A, remove the User B as approver and set the User A as approver.
5. See, that for User B in the web the expense disappeared in the LHN. It's cleaned out by the pusher update:

![image](https://github.com/user-attachments/assets/b55b239f-1035-43e7-9b80-5d1c39033b62)

6. As User B, go to the Search tab, the expense will be displayed there -> open it. After that it will start to be shown in the LHN on the web.
7. Open the ios app where we logged in as User B as well. There will be no expense in the LHN. This way the LHN on web and ios looks different.

**The video with all of the steps:**

https://github.com/user-attachments/assets/578877bf-1910-4afc-9cda-94fe0d741f19

**Comment by VickyStash** [comment]
@Beamanator Since you were working on the [related functionality](https://github.com/Expensify/App/pull/49557), could you please take a look at the [comment](https://github.com/Expensify/App/issues/51349#issuecomment-2437452929) above? I think the expense report shouldn't disappear from the LHN in this case and should be get by the app if user opens the app.

**Comment by VickyStash** [comment]
**Update:** waiting for @Beamanator thoughts based on the [investigations](https://github.com/Expensify/App/issues/51349#issuecomment-2437452929) above

**Comment by VickyStash** [comment]
**Note:** I'm going to be OOO Oct 31-Nov 3 🌴

**Comment by Beamanator** [comment]
@VickyStash sorry but can you tell me the exact workspace settings (collect vs control, delayed submission, approval mode) before you started step 2 [here](https://github.com/Expensify/App/issues/51349#issuecomment-2437452929)?

**Comment by VickyStash** [comment]
@Beamanator 
![image](https://github.com/user-attachments/assets/41a32c45-a0ef-4808-b531-a2efff603a83)

1. Type: Collect
2. Delayed submission is off
3. Approval mode: Advanced

Note, I've added a small correction in [the step 2](https://github.com/Expensify/App/issues/51349#issuecomment-2437452929)

**Comment by Beamanator** [comment]
Thanks for the update @VickyStash !

Honestly I don't think `ADVANCED` approval mode should even be allowed for `TEAM` / Collect policies!! Right @garrettmknight @puneetlath ?? 😅 

@VickyStash can you please let me know the steps you took to get to that stage so we can get that fixed first?? 😅

**Comment by VickyStash** [comment]
> @VickyStash can you please let me know the steps you took to get to that stage so we can get that fixed first?? 😅

@Beamanator sure, in the new dot:
1. Create a new workspace.
2. Open More features and enable **Workflows** (turns policy `areWorkflowsEnabled` flag into `true`)
3. Go to the Workflows section.
4. Turn on Approvals toggle. At this stage we have BASIC approval mode.
5. Got to the Members section and add 2 more members.
6. Go to Workflows section.
7. Change approver to not admin member. <- this update turns approval mode in ADVANCED.

![image](https://github.com/user-attachments/assets/30347dc9-78d4-4cb9-9859-8ce54018134c)

**Comment by puneetlath** [comment]
> Honestly I don't think ADVANCED approval mode should even be allowed for TEAM / Collect policies!! Right @garrettmknight @puneetlath ?? 😅

Hm, yes, as far as I'm aware, that shouldn't be possible. I don't see the option to set that approval mode on Collect policies in the product. Do you?

**Comment by Beamanator** [comment]
> I don't see the option to set that approval mode on Collect policies in the product. Do you?

In NewDot you're right, I do not 🤔  I think right now the main options are for Collect:
1. Approvals off = Submit & Close
2. Approvals on = Submit & Approve

But maybe we accidentally allowed turning on approvals & setting advanced approvals for collect workspaces?

**Comment by VickyStash** [comment]
> Hm, yes, as far as I'm aware, that shouldn't be possible. I don't see the option to set that approval mode on Collect policies in the product. Do you?

@puneetlath @Beamanator As I mentioned [above](https://github.com/Expensify/App/issues/51349#issuecomment-2454448176), the approval mode automatically turns into ADVANCED for Collect policies if I set a non-admin user as an approver in the New Dot (API returns these updates).
So the user just changes the approver, and the mode updates automatically due to the mentioned change.

**Comment by Beamanator** [comment]
Ok I was planning to look into the fix myself, but it looks like there's a test [`testPolicyApprovalModeIsUpdatedWhenApprovalsAreUpdated`](https://github.com/Expensify/Auth/blob/4b1bb142cc2792109d2c43e49cc4792b1d454440/test/test/command/SetWorkspaceApprovalTest.cpp#L93) that specifically allows for this to happen 😅 

@tgolen can you please comment on if you believe this is supposed to be allowed (a Team / Collect policy on Advanced approvals)? AFAIK, Advanced approvals is a Control policy feature.

**Comment by tgolen** [comment]
When I added the `SetWorkspaceApproval` command, I only restricted it to paid policies, so that approvals could be set for either collect or control policies. Maybe that was wrong and it should only be restricted to control? I am not sure, but @JmillsExpensify should know.

**Comment by Beamanator** [comment]
bumped in slack - https://expensify.slack.com/archives/C06ML6X0W9L/p1731060374967819

**Comment by Beamanator** [comment]
Ok got confirmation in slack that `Advanced approvals are a Control feature.` 👍 

@tgolen do you mind working on that fix? Hopefully will be quick since you have context on those commands

**Comment by VickyStash** [comment]
We are waiting for the fix from the internal engineer. I guess @tgolen is going to take a look at [this problem](https://github.com/Expensify/App/issues/51349#issuecomment-2454448176).

**Comment by tgolen** [comment]
I just submitted a PR to fix the approval mode today. I'm guessing it will get deployed to production in 1-2 days.

**Comment by tgolen** [comment]
The PR is https://github.com/Expensify/Auth/pull/13108 and no, it hasn't been merged yet. It looks like it should merge today and then deploy tomorrow, Fri Nov 15

**Comment by VickyStash** [comment]
@tgolen Sorry, I don't have access to the PR you mentioned. Has it been deployed yet?

**Comment by tgolen** [comment]
No, it hasn't been merged or deployed yet.

**Comment by tgolen** [comment]
The PR has been merged now. It should be deployed on Monday if everything goes normally.

**Comment by muttmuure** [comment]
Still awaiting auth deploy I think

**Comment by puneetlath** [comment]
Will there be a front-end component to the fix here too? Or is the back-end PR all that's needed?

**Comment by tgolen** [comment]
I'm not sure. Once the backend fix has been deployed (it still is not deployed yet), I think we should re-evaluate the bug.

**Comment by VickyStash** [comment]
@tgolen were the updates deployed, should I retest the issue?

**Comment by tgolen** [comment]
It has been deployed now, yes. Can you please retest it?

**Comment by VickyStash** [comment]
Okay, so what I see right now:
1. The approval mode is Basic, even if I change the approver to non-admin user, so this is fixed ✅ The only thing I want to confirm is how it looks in the app (2 blocks are displayed in approvals, see the screenshot/video, is it expected)?
![image](https://github.com/user-attachments/assets/3562eebc-a2e4-4cfa-8e47-39d9b71f2e9a)

<details>
<summary>Video</summary>

https://github.com/user-attachments/assets/71455147-7c03-48a2-bc26-ff36b048c47c
</details>

---

2. If I change the approver, the expenses don't clean out from the LHN by the pusher for the previous approver anymore. ✅  

But, the expenses are not displayed in the LHN after re-login, no matter if I was removed as an approver or not.
They are shown only when I open them from the Search page.

<details>
<summary>Video</summary>
https://github.com/user-attachments/assets/1258e773-90d8-4e06-9d05-083e0db80b8f
</details>

It happens cause `OpenApp` API call doesn't provide the necessary report. It's received only when `OpenReport` for report on the Search page is called. 

<img width="1492" alt="2" src="https://github.com/user-attachments/assets/aa03d84d-745e-468b-ab96-d792154867b6">


<img width="1493" alt="1" src="https://github.com/user-attachments/assets/5b77b295-2878-4d09-9f23-ccbcf4675548">

**Comment by tgolen** [comment]
> The only thing I want to confirm is how it looks in the app (2 blocks are displayed in approvals, see the screenshot/video, is it expected)?

No, this is not expected. I am not able to reproduce this locally 🤔 so it must be some weird glitch you are experiencing. Can you try setting up the workspace fresh again and see if you can still reproduce it? See this video:

https://github.com/user-attachments/assets/97f7c324-daa5-45c8-bc3f-cc18f1bdbc97

> It happens cause OpenApp API call doesn't provide the necessary report. It's received only when OpenReport for report on the Search page is called.

Can you clarify what you mean when you say "OpenApp API call doesn't provide the necessary report"? What report are you expecting it to have?

> But, the expenses are not displayed in the LHN after re-login, no matter if I was removed as an approver or not.

I'm not sure what to make of the "no matter if I was removed as an approver or not" part.

Can you boil each of the problems down to a clear set of reproduction steps and an expected/actual outcome?

**Comment by VickyStash** [comment]
### Issue: 
Two approval flows are shown in the section if I change the approver to non-admin member.

### Notes:
User: `viktoryia.kliushun+new1@callstack.com`
Policy: `AF12F09BA5B3DCEC`

### Steps:
1. Create a new account.
2. Create a new workspace.
3. Turn on **Workflows** feature.
4. Add two members.
5. Go to Workflows page and turn on approvals toggle.
6. Update the approver to be non-admin member.
7. Wait a couple of seconds, see the extra block in the approvals section.


https://github.com/user-attachments/assets/d3c1e858-cc5e-420b-8500-6a846ab2175b

### Reason for showing the second block
**In my example:**
`viktoryia.kliushun+new1@callstack.com` - workspace admin
`viktoryia.kliushun+2@callstack.com` - non admin member, which was assigned as an approver

After I set the `viktoryia.kliushun+2@callstack.com` as an approver , the `approver` field inside policy changes to `viktoryia.kliushun+2@callstack.com` as well.
But inside `employeeList` every user submits to `viktoryia.kliushun+new1@callstack.com`.

![image](https://github.com/user-attachments/assets/7538a1bf-2b9c-444f-a61c-99285cc2a21f)

So the app adds a second block:
https://github.com/Expensify/App/blob/2166dffba5b3f0283d5cf2d9f0b700749c8fdcf1/src/libs/WorkflowUtils.ts#L163-L171

**Comment by VickyStash** [comment]
### Issue:
Non-admin approver doesn't see expense to approve in LHN after re-login.
 
### Steps:
We will have 3 users: **Admin**, **Submitter**, **Approver**
1. Login as Admin, create a workspace.
2. Turn on workflows feature.
3. Add Submitter and Approver as members of the workspace.
4.  Turn on approvals on the workflow page. Set Approver **(non-admin user)** as the approver.
5. Login as Approver on one device and as a Submitter on another one.
6. As a Submitter - submit the expense inside a workspace.
7. As the Approver - see that the expense appeared in your LHN:
<img width="1314" alt="Expense1" src="https://github.com/user-attachments/assets/01edf713-017f-419d-a572-b6478f7eb64d">
8. Log. out from Approver account and login again, see that **no expense in LHN is displayed anymore**.

https://github.com/user-attachments/assets/7db2485e-a5c3-41c2-adae-536498d21565

**Note:** if you go to the Search page and open expense there - the expense will start to be displayed in the LHN as well. You can see it in the video above

### Reason for not showing expense report to approver after re-login
The Expense Report that I expect to see as `viktoryia.kliushun+d@callstack.com` user is the one with the id:  `230002486694357` (you can see it on the screenshot above) in the step 7.

But I don't get it inside `OpenApp` API command:
![image](https://github.com/user-attachments/assets/c31c3735-3efc-4865-9e4c-ec9f86795729)

Due to this it's not displayed in the LHN after re-login.

**Comment by VickyStash** [comment]
@tgolen I've added step by step description of both issues above, let me know if you need anything else!

**Comment by tgolen** [comment]
Thanks, I think that first one has already been fixed, but maybe it just hasn't deployed yet. I'm going to wait until next week to see if I can reproduce it in production.

@garrettmknight I know you've been working on polishing these flows too. Have you seen the problem Vicky reported [here](https://github.com/Expensify/App/issues/51349#issuecomment-2494088943)?

I am a non-admin approver on our Expensify-US policy, and I haven't experienced this I don't think. If I log out and log back in, I still see the reports I need to approve in my LHN. I don't have any reports to approve right now, but when I do, I'm going to try and reproduce this. This one sounds difficult to debug, but reproducing it is a start.

**Comment by tgolen** [comment]
OK, I just had a report to approve, so I logged out, logged back in, and it was still showing in my LHN 🤔 so there must be something else going on...

**Comment by tgolen** [comment]
I will attempt to reproduce it in my local environment today according to your steps and report back!

**Comment by tgolen** [comment]
# Daily Update
- I was able to reproduce this locally. So, that's the first step! 
- I have absolutely no idea what causes it

### Next Steps
- @tgolen dig into `OpenApp` locally to debug why the report isn't being returned

#### ETA
- I ran out of time today, so I will probably start this tomorrow and have a better ETA once that is done.

**Comment by tgolen** [comment]
# Daily Update
- I tracked down where this is happening and it's due to this [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/44277) from @marcaaron a couple of weeks ago
- I'm unsure of what solution we want for this and [we are discussing it in Slack](https://expensify.slack.com/archives/C03TQ48KC/p1732641559055219)

### Next Steps
- @tgolen @marcaaron figure out a good path forward that fixes this case
- Implement the fix and get it reviewed and merged

#### ETA
- Monday Dec 2

**Comment by tgolen** [comment]
# Daily Update
- There was quite a bit of [discussion in Slack](https://expensify.slack.com/archives/C03TQ48KC/p1732641559055219) about this.
- https://github.com/Expensify/Expensify/issues/448335 was opened to try and track down and fix what we think the actual bug is (a bug in step 4 of [these steps](https://github.com/Expensify/App/issues/51349#issuecomment-2494088943) where the workspace chat isn't being shared with the updated approver)

### Next Steps
- I think this issue can be placed on HOLD until https://github.com/Expensify/Expensify/issues/448335 is fixed and then this can be re-evaluated

#### ETA
- TBD

**Comment by tgolen** [comment]
# Daily Update
- Still on hold with https://github.com/Expensify/Expensify/issues/448335 See this issue for updates

**Comment by tgolen** [comment]
# Daily Update
- Same

**Comment by tgolen** [comment]
# Daily Update
- I'm changing this to weekly priority for now

**Comment by tgolen** [comment]
# Weekly Update
- I am still chugging away at this and the current PR I am working on is https://github.com/Expensify/Auth/pull/13442
- It should be done reviewing soon and merged

### Next Steps
- @tgolen get that PR merged and deployed
- Then we can re-evaluate what's going on with the LHN and see if there are any more bugs

#### ETA
- Wednesday, Dec 18

**Comment by VickyStash** [comment]
Hey @tgolen, any updates?

**Comment by tgolen** [comment]
Yeah, I usually update my weekly issues on Friday, but I am feeling festive and can give you an early update!

# Weekly Update
- The final backend PR https://github.com/Expensify/Auth/pull/13442 was just merged today

### Next Steps
- Once it has been deployed, this bug can be re-evaluated to see if it's fully fixed or what else needs to be done to close it out.

#### ETA
- Tomorrow, Dec 20

**Comment by tgolen** [comment]
The backend PR was deployed yesterday. @VickyStash Can you please give this another look and see if it's ready to be closed out?

**Comment by VickyStash** [comment]
Thank you! I'll re-test it on Monday morning

**Comment by VickyStash** [comment]
Results of testing:

**User A** - admin of the workspace
**User B** - non-admin user, approver
**User C** - expense submitter

✅  Workspace approval mode turns into _Advanced_ if non-admin User B is selected as the approver - **FIXED**
✅  Expense doesn't disappear from LHN for User B if he is not approver anymore -  **FIXED**

➖ Expense for approval isn't display for User B if he is not approver anymore and re-login the app.

**Steps:**
1. From User A 
1.1 create a workspace
1.2 add User B and User C as members
1.2 enable Workflow feature
1.3 turn on approvals and make the User B approver.
2. From User C submit the expense to the workspace.
3. See that for User B the expense appeared in the LHN.
4. From User A, remove the User B as approver and set the User A as approver.
5. See that User B still has expense in the LHN.
6. **Re-login as User B. See that the expense isn't displayed in the LHN**.
7. Go to the Search page, open the expense from there. Go back to the main tab . **See that now the expense is displayed in the LHN.**

https://github.com/user-attachments/assets/ddb9085e-9013-4aab-8ce2-d5d037dd72e5

**Comment by VickyStash** [comment]
I'm going to be OOO Dec 25 - Dec 29 🎄

**Comment by tgolen** [comment]
> ➖ Expense for approval isn't display for User B if he is not approver anymore and re-login the app.

@youssef-lr @aldo-expensify I think we discussed this one. Is the previous non-admin approver supposed to retain access to the reports? In my PR, we unshared the reports from the previous non-admin approver I thought.

https://github.com/Expensify/Auth/blob/2a9f31e420a09bdb40f335d64a6be62641eff86e/auth/lib/Policy.cpp#L2596-L2603

Should we remove that, or update the testing steps because that is now expected that they would be unshared?

**Comment by youssef-lr** [comment]
I think the previous approver loses access to the reports, however IIRC we decided if a previous approver still has reports to approve, they should retain access to the expense reports. I think @garrettmknight or @trjExpensify can confirm.

**Comment by aldo-expensify** [comment]
> I think the previous approver loses access to the reports, however IIRC we decided if a previous approver still has reports to approve, they should retain access to the expense reports. I think @garrettmknight or @trjExpensify can confirm.

This is my understanding too. The previous approver only retains access if they are currently the `report.managerID`.

**Comment by garrettmknight** [comment]
Agree with @youssef-lr and @aldo-expensify

**Comment by tgolen** [comment]
OK, thanks everyone! @VickyStash, does the expected behavior that they pointed out make sense and can you verify if that is the actual behavior?

**Comment by VickyStash** [comment]
I haven't had enough time today for this one, will take a look tomorrow!

**Comment by VickyStash** [comment]
> OK, thanks everyone! @VickyStash, does the expected behavior that they pointed out make sense and can you verify if that is the actual behavior?

@tgolen That's true, my User B (non-admin user, previous approver) still has access to the expense as he is current `report.managerID`.

**But after fresh login, the expense is not displayed in the LHN.**

I can access it only from Search -> Expenses page. If I open the expense from Search -> **it starts to be displayed in the LHN**.

Please, see the test steps and the recording reported in this comment: https://github.com/Expensify/App/issues/51349#issuecomment-2559265441

So it's not about expense access problem, but more about missing this expense inside `OpenApp` API call.

**Comment by tgolen** [comment]
OK, thank you. Those details are clear enough that I should be able to look
into that specifically. I'll let you know what I find out.

**Comment by tgolen** [comment]
@aldo-expensify or @youssef-lr can you please look into this and see what you're able to find?

When I followed the testing steps, I was able to reproduce the issue. but...
1. The report has no `managerID` in the reports table (which kind of contradicts what was said above)
2. If I insert User B's account ID as the `managerID` in the reports table, the report is still not returned in `OpenApp`

**Comment by youssef-lr** [comment]
@tgolen I'm unable to reproduce, after clearing cache and restarting I can still see the expense report

https://github.com/user-attachments/assets/b41ef170-c46a-40bb-87f5-d6ff7c01f6f3

**Comment by youssef-lr** [comment]
@tgolen this doesn't seem implemented in `getChats`? we seem to only fetch the _parent_ reports of expenses waiting for approval from the current user, and not the expense reports themselves: https://github.com/Expensify/Auth/blob/14f48f2de74b85ac32c1df76c2d592b2d08cb9a6/auth/lib/Report.cpp#L8394-L8423

**Comment by youssef-lr** [comment]
I really don't understand why adding this condition below [here](https://github.com/Expensify/Auth/blob/14f48f2de74b85ac32c1df76c2d592b2d08cb9a6/auth/lib/Report.cpp#L8353) didn't work:

```
"    OR (r.details->>" + SQ(NVP_DETAILS_TYPE) + " = " + SQ(TYPE_TASK) + " AND r.managerID = " + SQ(accountID) + " AND status = " + SQ(STATUS_OPEN) + ") "
```

**Comment by VickyStash** [comment]
Hey @youssef-lr , any updates on this one?

**Comment by tgolen** [comment]
bump @youssef-lr for any updates?

**Comment by tgolen** [comment]
bump @youssef-lr ?

**Comment by youssef-lr** [comment]
Sorry @tgolen I was OOO sick. I've investigated this again today but haven't figured it out yet, the GetChats query is a bit complex to debug. I'm still on it, hopefully I'll figure it out by Monday.

**Comment by youssef-lr** [comment]
Posted an update here https://expensify.slack.com/archives/C05R2V5GM1S/p1738121135867459?thread_ts=1730142545.830059&cid=C05R2V5GM1S

**Comment by tgolen** [comment]
I'm reassigning this to @youssef-lr while he digs into this final problem. Sorry @VickyStash I didn't mean to unassign you!

**Comment by tgolen** [comment]
It sounds like @cristipaval is able to pick this up and discovered the root cause so I will reassign it to him and a PR should be ready in a day or so.

**Comment by cristipaval** [comment]
Update:

- Draft [Auth PR](https://github.com/Expensify/Auth/pull/14156) is up with the fix. Waiting for an early approval before writing the automated tests and query timing
- [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/45487) is ready for review and merge, if the Auth PR gets the green light

**Comment by cristipaval** [comment]
Update:

- [Auth PR](https://github.com/Expensify/Auth/pull/14156) is ready for review
- [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/45487) is approved. I'll merge it when when Auth PR passes the query timing

**Comment by cristipaval** [comment]
### Weekly Update
 - [Auth PR](https://github.com/Expensify/Auth/pull/14156) is waiting for the final approval. The query timing was not great initially, but @nkuoch helped improve it 🙇 
 - [Web-E](https://github.com/Expensify/Web-Expensify/pull/45487) has been merged and waiting to be deployed.

**Comment by cristipaval** [comment]
[Auth PR](https://github.com/Expensify/Auth/pull/14156) has been deployed. And [Web-E](https://github.com/Expensify/Web-Expensify/pull/45487) is in staging. @tgolen, is there any work left on the front-end here?

**Comment by tgolen** [comment]
Not that I know of!

**Comment by cristipaval** [comment]
Web-E PR is in production, closing the issue.
